pkg/controllers/management/etcdbackup/etcdbackup.go

```diff
@@ -112,7 +109,18 @@ func (c *Controller) Remove(b *v3.EtcdBackup) (runtime.Object, error) {
 	if b.Spec.BackupConfig.S3BackupConfig == nil {
 		return b, nil
 	}
-	return b, c.deleteS3Snapshot(b)
+	// try to remove from s3 for 3 times, then give up. if we don't we get stuck forever
+	var delErr error
+	for i := 0; i < 3; i++ {
+		if delErr := c.deleteS3Snapshot(b); delErr != nil {
```

Comment:
I think we should omit if err is NotFound

Reply:
It's handled internally (or not) by the s3 library:
```
	// DeleteObject always responds with http '204' even for
	// objects which do not exist. So no need to handle them
	// specifically.
        return nil
```